### ISSUE TITLE
Bluemix one click deploy failing in deploy stage build successful

### ISSUE THREAD

**Issue by daminisatya** [loklak/loklak_server#318]
Same errors as in : https://github.com/loklak/loklak_server/pull/316#issuecomment-216429691

**Comment by jig08** [comment]
I believe the reason could be missing `declared services:` in the `manifest.yml` file.
We need to mention the optional services that are expected to be set up before the app is deployed.
Could you please help me list 'em out ?

@Orbiter @daminisatya @sudheesh001

**Comment by Orbiter** [comment]
I cannot find a hint about which service is missing, do you have more error logs?

**Comment by jig08** [comment]
@Orbiter My `IBM Bluemix` free trial period \*just* expired :cry: 
Now the `Project Creation` only fails at the first step.

**Comment by sudheesh001** [comment]
@jig08 Since the build works fine I don't think there's a problem with any of the dependencies here, the issue is in running the start as far as i can guess.

**Comment by jig08** [comment]
@sudheesh001 I tried exploring a bit, but no luck so far.
Will read more about stage's run-time information, maybe.

**Comment by sudheesh001** [comment]
@jig08 You might need a `.pipeline.yml` file i guess for the stages

**Comment by jig08** [comment]
@sudheesh001 Probably yes. We need to add the `pipeline.yml` in the `.bluemix` directory.

Should we create one manually ? Or if you have a valid Bluemix account (mine just expired a few days back), you could generate one from the existing project there.

Ref [Creating a Deploy to Bluemix button](https://console.ng.bluemix.net/docs/develop/deploy_button.html)

**Comment by sudheesh001** [comment]
Build succeeded but deploy failed again @shivenmian

**Comment by shivenmian** [comment]
Logs?

**Comment by shivenmian** [comment]
Okay @sudheesh001 @jig08 

I checked the logs. There was a problem in the buildpack syntax (we were missing the applications: tag). I will add it. But still when I added it locally it wouldn't deploy, because of the same staging error.

I'm definitely sure we need a pipeline.yml because that's what defines the stages. I'll work on it and submit a PR shortly.

**Comment by jig08** [comment]
@shivenmian Yes, as mentioned https://github.com/loklak/loklak_server/issues/318#issuecomment-217609641 we could create `pipeline.yml` either manually or generate one from the existing project in Bluemix.

If you have a valid account, pls go ahead to include an auto-generated one.

Reference: [Creating a Deploy to Bluemix button](https://console.ng.bluemix.net/docs/develop)/deploy_button.html

**Comment by leonmak** [comment]
I've tried a few things to get the yaml file:
- viewing the pipeline after deploy failed, in the build stage I removed "output" in the "build archive directory" so the whole thing would be passed to deploy stage. there is no output folder after build so that's why #316 had input not found.
- after that in deploy stage, the logs said that JAVA_HOME was not defined correctly, so I looked at other buildpacks and saw that some installed java 1.7, but the deploy was timing out and could not start.
- luckily I found some slides that said to add "sleep 1d" in the command for cf push.
- the routes are set to us-south as any other region gets a route error on clicking "view my app"

Now search is working but the application logs say that some dependencies like for log4j are missing
